### ISSUE TITLE
FileType.custom, allowedExtensions:[] does not work on Huawei Mate 30 Pro

### ISSUE THREAD

**Issue by Thioby** [miguelpruivo/flutter_file_picker#312]
Setting `FilePicker.getFile(type: FileType.custom, allowedExtensions: ["pdf", "jpg", "png", "doc", "docx", "xls", "xlsx"]);`
does not restrict MIME types just to allowed one. Instead of this, Huawei's  system file picker allow select any type of file.

Android 10 EMUI 10.1

Quick dirty hot fix: 
In `FilePickerDelegate.java` line 188: change: `intent = new Intent(Intent.ACTION_GET_CONTENT);` to  ` intent = new Intent(Intent.ACTION_OPEN_DOCUMENT);`

**Comment by miguelpruivo** [comment]
Open document behaves different than get content and it shouldn't be used for this purpose due to its own limitations. I'm surprised it worked for with with open document instead of get content.

I'll have to investigate.

**Comment by leszekkrol** [comment]
@miguelpruivo I got similar problem with picking files. On the same device, downloading images larger than 4096 px wide damages the file :(

**Comment by miguelpruivo** [comment]
@leszekkrol can you try using the `fix/open_document` branch directly and tell me if that fixes it for you?

Thanks!

**Comment by leszekkrol** [comment]
@miguelpruivo I will check this branch on Monday :-)

**Comment by leszekkrol** [comment]
@miguelpruivo I got few minutes to check it today, unfortunately still after selecting the file on android 10, the file remains corrupted. after selecting a file I am not even able to display it. 

the error applies to both Huawei Mate 30 Pro and Huawei Mate 20 Pro.
More information about device: https://ibb.co/wg5VXsF

**Comment by miguelpruivo** [comment]
Does that happen only with Huawei devices?

**Comment by leszekkrol** [comment]
it looks like it, but the question is, can anyone else confirm it? From what I noticed on Samsung with android 10, the application works flawlessly.

**Comment by miguelpruivo** [comment]
Huawei devices have been raising a lot of issues lately, I don't really get why.

**Comment by petroniuchacz** [comment]
@miguelpruivo I tried to use it on Pixel 2 Android 8.1 and Pixel 2 Android 10.0 emulators and on Xiaomi Redmi 6A MIUI 10.3.8, it doesn't seem to work on any of these.
![image](https://user-images.githubusercontent.com/24723590/89340100-7e941700-d69f-11ea-81bc-d8dc365a11aa.png)

**Comment by miguelpruivo** [comment]
@petroniuchacz what do you mean? What does actually don't work?

**Comment by petroniuchacz** [comment]
@miguelpruivo I'm able to pick any file type. On the provided picture I wanted to restrict to mp4, but I was able to choose jpg.
Or didn't I get what the feature is for?

**Comment by petroniuchacz** [comment]
I double checked - it actually doesn' work if I do Open file picker > drawer with source options > **Photos** . If I go Open file picker > drawer with source options > **any other option**, the restriction holds.

**Comment by miguelpruivo** [comment]
@petroniuchacz unfortunately that’s not “an issue per se”, what’s happening with you is described [here](https://github.com/miguelpruivo/flutter_file_picker/wiki/Troubleshooting#-issue-1) on the troubleshooting area.

Tl;dr: you should always validate the picked files on your end and discard/present an error to the user if you plan to do it on Android, as it’s up to the pickers/providers to honour the filters provided.
